### PR TITLE
=htc,doc #18535 improved docs on spray-json usage

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/18535

I also increased the max depth of the top-level TOC for streams and http. While this is a crude solution, it already helps a lot with discoverability of docs (I can finally `Ctrl+F "json"`, which previously required digging "oh, because it's _common_"). Proper fixing of the search bar (I do remember about it) will properly address this issue.

### CODE REVIEW

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/SprayJsonExampleSpec.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    //#example
+    import spray.json._
+
+    // domain model
+    final case class Item(name: String, id: Long)
+    final case class Order(items: List[Item])
+
+    // collect your json format instances into a support trait:
+    trait JsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
+      implicit val itemFormat = jsonFormat2(Item)
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/SprayJsonExampleSpec.scala]: add a comment where jsonFormatX comes from

File: akka-docs-dev/rst/scala/http/common/json-support.rst [hunk 2]
```diff
@@ -21,7 +21,7 @@ that an implicit ``spray.json.RootJsonReader`` and/or ``spray.json.RootJsonWrite
 
 This is how you enable automatic support for (un)marshalling from and to JSON with `spray-json`_:
 
-1. Add a library dependency onto ``"com.typesafe.akka" %% "akka-http-spray-json-experimental" % "1.x"``.
+1. Add a library dependency onto ``"com.typesafe.akka" %% "akka-http-spray-json-experimental" % "@version@"``.
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/http/common/json-support.rst]: what is this variable resolving to?

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/SprayJsonExampleSpec.scala (hunk 1 above)
Comment (ktoso) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/SprayJsonExampleSpec.scala]: good one, I'll add that comment in another PR then :)

File: akka-docs-dev/rst/scala/http/common/json-support.rst (hunk 2 above)
Comment (ktoso) [inline akka-docs-dev/rst/scala/http/common/json-support.rst]: the version of streams, so "`1.0`"

**Pull request merged**